chopper/lib/src/base.dart

```diff
@@ -171,23 +171,23 @@ class ChopperClient {
     if (service == null) {
       throw Exception('Service of type \'$serviceType\' not found.');
     }
-    return service;
+    return service as ServiceType;
   }
 
-  Future<Request> _encodeRequest(Request request) async {
+  Future<Request?> _encodeRequest(Request request) async {
     if (converter != null) {
-      return converter.convertRequest(request);
+      return converter?.convertRequest(request);
     }
 
     return request;
   }
 
   Future<Response<BodyType>> _decodeResponse<BodyType, InnerType>(
```

Comment:
It's not possible in Dart. next compile time error raised

```
'throw' can't be used as an identifier because it's a keyword.
Try renaming this to be an identifier that isn't a 
```

Reply:
Fair enough, keep it as-is.